docs/README.md

```diff
@@ -0,0 +1,155 @@
+| status.canary       | StatusConfigurationSpec         | The status of the configuration including name, replicas, and traffic. |
+| status.conditions   | List\<[Condition](#Conditions)> | The name for your Service. This will ultimately derive DNS name. |
+
+## Understanding The Data Model
+KFService offers a few high level specifications for common ML technologies. These come out of the box with a performant model server and a low flexibility, low barrier to entry experience. For more complex use cases, users are encouraged to take advantage of the custom spec, which allows a user's container to take advantage of the KFService architecture while providing the flexibility needed for their use case. 
+
+### ModelSpec
+| Field       | Value       | Description |
+| ----------- | ----------- | ----------- |
+| spec.[default,canary].tensorflow  | [Tensorflow](#Tensorflow)   | A high level specification for Tensorflow models. |
+| spec.[default,canary].xgboost     | [XGBoost](#XGBoost)         | A high level specification for XGBoost models. |
+| spec.[default,canary].scikitlearn | [ScikitLearn](#ScikitLearn) | A high level specification for ScikitLearn models. |
+| spec.[default,canary].pytorch     | [Pytorch](#Pytorch)         | A high level specification for Pytorch models. |
+| spec.[default,canary].custom      | [Custom](#Custom)           | A flexible custom specification for arbitrary customer provided containers. |
+
+### Rollouts
+The traffic management rules are specifically targetting Rollout use cases. A/B Testing, Multi-Armed-Bandit, and Ensemble Inferencing should not be implemented using KFService's default/canary mechanism. More complex controllers for these features should be purpose built and operate at a higher level than a KFService, perhaps even implemented using multiple KFServices or a KFGraph. This decoupling is critical to enable features like canarying a change to a Multi-Armed-Bandit inferencing graph.
+
+#### Blue Green Rollout
+The simpliest deployment strategy relies on KFService's default spec. The rollout workflow is identical to a Kubernetes Deployment. Changes to the CRD will create a new set of serving pods. Once the resources are available for serving, traffic will be flipped to the new resources. Users may observe the status of the KFService to gain insight into the current state of the Blue Green rollout.
+
+#### Pinned Rollout
+For more cautious and manual workflows, users may follow the Pinned Release pattern. It is executed using the following:
+
+* Start with a KFService that has a "stable" or "golden" default definition.
+* Modify the KFService to set the new ModelSpec under the `canary`.
+* Set trafficPercent to 0
+* Observe the KFService until the new revision progresses
+    * Note: traffic is still sent to "default"
+* Execute any inspection or testing against `canary`
+    * Note: The revision will be available under a specialized URI (e.g. revisionName.name.namespace.cluster.internal)
```

Comment:
would the canary would also be available in a revision independent URI? e.g. name.namespace.canary.cluster.internal

Reply:
Yes! I've cleaned this up.